File: Client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams{
    gender:string;
    minAge=1;
    maxAge=100;
    pageNumber=1;
    pageSize=5;
    orderBy="lastActive";

    constructor(user:User) {
    this.gender=user.gender==='female' ?'male' : 'female';

    }
}

File: API/Helper/Userparams.cs
﻿namespace API.Helper
{
    public class Userparams :Paginationparams
    {
       
        public string? CurrentUsername { get; set; }
        public string? Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;
        public string? OrderBy { get; set; } = "lastActive";

    }
}
